internal/pkg/api/handleCheckin.go

```diff
@@ -295,12 +295,14 @@ func (ct *CheckinT) writeResponse(zlog zerolog.Logger, w http.ResponseWriter, r
 		}
 	}
 
-	span, _ := apm.StartSpanOptions(r.Context(), "action delivery", "fleet-server", apm.SpanOptions{
-		Links: links,
-	})
-	span.Context.SetLabel("action_count", len(fromPtr(resp.Actions)))
-	span.Context.SetLabel("agent_id", agent.Id)
-	defer span.End()
+	if len(fromPtr(resp.Actions)) > 0 {
```

Comment:
this is just moving the trace into a condition to reduce noise ("action deliver" span was showing up even though there were no actions)